### DISCUSSION TITLE
HistoricalStockPriceDataUpdater가 없어졌나요?

### DISCUSSION THREAD

**Discussion by Card2Me** [elbakramer/koapy#28] [Q&A]
올해 초에 koapy를 잘 사용했는데 아래처럼 HistoricalStockPriceDataUpdater 이 있었습니다.

그런데 git 소스에도 HistoricalStockPriceDataUpdater 이 없네요?

아시는분 계셔요?

**Comment by elbakramer** [comment]
아마도 아래 코드를 말씀하시는거 같은데요.
https://github.com/elbakramer/koapy/blob/v0.1.15/koapy/data/HistoricalStockPriceDataUpdater.py

제가 0.1.15 버전 이후로 없엔거로 보이네요.
당시에 제가 해당 기능을 잘 안쓰기도 했고, 핵심 피쳐는 아니라고 생각해서 과감하게 없에버렸던거 같습니다.

기능을 잘 안썼던 이유는, 당시에 종목별로 엑셀파일 형태로 데이터를 저장을 해봤었는데요. 이게 사람이 눈으로 확인하는 것은 편했지만, 나중에 뭔가 대량으로 백테스팅등에 사용하려고 보니까 읽어오는 속도가 너무 느리더라고요. 키움 API 속도 자체도 느리기도 했고..

그리고 사실 맨 처음에는 키움 기반 일봉데이터만 생각하고 만들었다가 분봉, 사이보스 관련 지원도 약간은 억지로 집어넣으면서 코드랑 객체 자체가 너무 복잡해진 감도 없지 않아 있어서요. 이후에 유지보수가 어렵지 않을까 하는 생각도 한 몫 했던거 같습니다.

코드 내용 자체는 사실상 위에 링크된 스크립트 하나에 거의 다 들어있어서, 만약 계속 사용하시려면 위의 코드를 따로 받은 뒤 약간의 수정을 거쳐서 사용하시면 어떨까 싶습니다.

잠깐 살펴보니 아래 세가지 관련해서 정도만 바꾸면 될거 같아 보입니다.

```python
# from koapy import KiwoomOpenApiContext
from koapy.backend.kiwoom_open_api_plus.core import KiwoomOpenApiPlusEntrypoint

# from koapy.backend.cybos.CybosPlusComObject import CybosPlusComObject
from koapy.backend.daishin_cybos_plus.core import CybosPlusEntrypoint

# from koapy.utils.krx.calendar import get_last_krx_close_datetime
def get_last_krx_close_datetime():
    from pandas import Timestamp
    from exchange_calendars import get_calendar
    krx_calendar = get_calendar('XKRX')
    now = Timestamp.now(tz=krx_calendar.tz)
    last_close = krx_calendar.previous_close(now).astimezone(krx_calendar.tz)
    return last_close
```

**Reply by elbakramer** [reply]
기존의 `HistoricalStockPriceDataUpdater`  와 비슷한 성격으로 현재는 (0.4.0 버전 기준) 아래가 있긴 한데요. 
https://github.com/elbakramer/koapy/blob/v0.4.0/koapy/utils/data/KrxHistoricalDailyPriceDataLoader.py

질문자분의 사용성과 맞을지는 잘 모르겠네요.
일단 참고 차원에서 공유드립니다.

**Reply by Card2Me** [reply]
덕분에 https://koapy.readthedocs.io/en/latest/notebooks/getting-historical-stock-price-data.html 도 읽어보게 되었습니다.

15분데이터가 필요해서, 0.1.15버젼의 소스를 가져다 써야 겠네요.

자세한 설명 감사드립니다.
